integrated_channels/moodle/client.py

```diff
@@ -36,31 +66,43 @@ def __init__(self, enterprise_configuration):
         """
         super(MoodleAPIClient, self).__init__(enterprise_configuration)
         self.config = apps.get_app_config('moodle')
+        self.token = enterprise_configuration.token or self._get_access_token()
 
+    @moodle_request_wrapper
     def create_course_completion(self, user_id, payload):  # pylint: disable=unused-argument
         pass
 
+    @moodle_request_wrapper
     def delete_course_completion(self, user_id, payload):  # pylint: disable=unused-argument
         pass
 
+    @moodle_request_wrapper
     def get_course_id(self, key):
         """
         Gets the Moodle course id (because we cannot update/delete without it).
         """
         params = {
-            'wstoken': self.enterprise_configuration.api_token,
+            'wstoken': self.token,
             'wsfunction': 'core_course_get_courses_by_field',
             'field': 'shortname',
             'value': key,
             'moodlewsrestformat': 'json'
         }
         response = requests.get(
-            self.enterprise_configuration.moodle_base_url,
+            urljoin(
+                self.enterprise_configuration.moodle_base_url,
+                self.MOODLE_API_PATH,
+            ),
             params=params
         )
+        parsed_response = json.loads(response.text)
+        if not parsed_response.get('courses'):
+            raise ClientError('MoodleAPIClient request failed: 404 Course key '
+                              '"{}" not found in Moodle.'.format(key))
 
-        return json.loads(response.text)['courses'][0]['id']
+        return parsed_response[0]['id']
```

Comment:
Might want to add some handling around array length and key existence. You'll also not want the moodle_request_wrapper on this if you're not returning a raw response